### DISCUSSION TITLE
Full support for soft deletes?

### DISCUSSION THREAD

**Discussion by Jcalcaldev** [mikro-orm/mikro-orm#5281] [General]
I've been trying to implement soft deletes for a couple of days no with no success.

The closest I've gotten to success is trying https://github.com/TheNightmareX/mikro-orm-soft-delete 

Filtering works as intended, and rows are soft-deleted as intended. The issues arise with related entities. I've tried with entities marked with `deleteRule: cascade` and `orphanRemoval: true`, with no success. They're not updated.

```
  @ManyToOne(() => User, {
    inversedBy: 'userBoundAssets',
    nullable: true,
    deleteRule: 'cascade',
  })
  public boundToUser?: User;
  ```
  
This is just an example. Deleting the User row would update its `deletedAt` property, but this Asset would remain exactly the same, with its `bound_to_user_id` column still pointing to the soft-deleted User and its `deletedAt` property remaining as `null`.

There are also OneToOne and OneToMany relations. None of them work.

Is it currently planned to implement soft deletes somewhere in the future? Maybe I'm missing something? Has anyone managed to implement them with relations-support?

**Comment by B4nan** [comment]
Since v6, to-one relations with filters on the target entity will cause implicit join of the target with filters applied, using soft deletes wont make this FK removed, but when you query for the entity its value should become `null` if the target is soft deleted (or better say, when it has a filter which makes it non-existent).

Note that `deleteRule` has (almost) no effect on runtime, it's a schema flag (and the operation is triggered by the database engine, not by the ORM). Orphan removal is quite a different thing, one controls what happens if you remove the owning entity (the one that defines the FK) and the other is about the target (what happens with the target if you remove the owner). None of that will have any effect on filters, filters are about selecting, and those options are about modifying/flushing. 

It sounds to me like all you need is to populate your relations, as that is the only way to make orphan removal triggered, which in turn will try to delete the relation, which results in the update. You need the relations to be populated for orphan removal to work. Note that there is no orphan removal support for m:1 relations, a `cascade: all` would do it for those - but that's quite a dangerous setting for a m:1 relation, you can end up removing staff you actually don't want to remove (things that are referenced from other entities, which in combination with `updateRule: cacade` means you would automatically remove also all the entities that reference the same target - this is btw the reason why there is no support for orphan removal on m:1, you'd have to query the database first to see all the references to the same target).

So provide a complete repro and then we can talk, maybe there is a room for improvement.

**Reply by Jcalcaldev** [reply]
It appears populating relations was what was missing, thanks for the explanations 🔥
